src/execution/index.ts

```diff
@@ -88,14 +89,16 @@ async function sendTransactionInternal(
   initialTokenRequests: TokenRequest[],
   signers?: SignerSet,
 ) {
-  if (sourceChain) {
-    const isAccountDeployed = await isDeployed(sourceChain, config)
+  // if source chain is provided, check if account is deployed on that chain.
+  // if source chain is not provided, it's a same chain transaction -> check if account is deployed on target chain (sourceChain === targetChain)
+  const fromChain = sourceChain ?? targetChain
+  if (fromChain) {
+    const isAccountDeployed = await isDeployed(fromChain, config)
```

Comment:
This change ensures that if sourceChain is undefined, and only targetChain is provided, we still check if the account is deployed on the relevant chain. 
Previously, the check was skipped in such cases, leading to failures when the wallet wasn't yet deployed on the target chain. Now, deployment is verified for both cases.